### PR TITLE
Refactor Ruby/Python Comment fixers

### PR DESCRIPTION
**Pull request by geigerj**
Ruby and Python use similar regular expressions to make proto
comments idiomatic. Factor these regexes out into a utility class.

### CODE REVIEW

**Comment by geigerj** [comment]
Fixes #40 
Also affects #38

File: src/main/java/com/google/api/codegen/CommentPatterns.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.google.api.codegen;
+
+import java.util.regex.Pattern;
+
+/**
+ * Abstract base class for formatting source comments to follow language-idiomatic style.
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/CommentPatterns.java]: This isn't abstract, or a base class

Comment (geigerj) [inline src/main/java/com/google/api/codegen/CommentPatterns.java]: Whoops, bad comment. Fixed.

Comment (geigerj) [inline src/main/java/com/google/api/codegen/CommentPatterns.java]: (FWIW: I had originally considered making the methods of `RDocCommentFixer` and `PythonSphinxCommentFixer` nonstatic and deriving them from an abstract base class, but I don't know that there's any benefit to doing so -- that might just be an unnecessary constraint.)

Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/CommentPatterns.java]: In general, I prefer to keep things more modular, and static utility classes are much more modular than abstract base classes.

**Comment by geigerj** [comment]
Note: added distinction between relative and absolute markdown links to fix an issue in #103.

**Comment by jmuk** [comment]
LGTM

**Pull request merged**